### PR TITLE
[API] add User Entity 

### PR DESCRIPTION
**Pull request by Dominskyi93**

### CODE REVIEW

File: src/main/java/com/example/carsharingservice/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public User get(Long id) {
+        return userRepository.getReferenceById(id);
+    }
+
+    @Override
+    public User findByEmail(String email) {
+        return userRepository.findByEmail(email);
```
Comment (AleksandrLts) [inline src/main/java/com/example/carsharingservice/service/impl/UserServiceImpl.java]:
```suggestion
        return userRepository.findByEmail(email)orElseThrow(
        () -> new EntityNotFoundException(custom message)
        );
```

File: src/main/java/com/example/carsharingservice/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        userFromDb.setEmail(user.getEmail());
+        userFromDb.setRole(user.getRole());
+        userFromDb.setFirstName(user.getFirstName());
+        userFromDb.setLastName(user.getLastName());
+        return userRepository.save(userFromDb);
+    }
+
+    @Override
+    public User get(Long id) {
+        return userRepository.getReferenceById(id);
```
Comment (AleksandrLts) [inline src/main/java/com/example/carsharingservice/service/impl/UserServiceImpl.java:28]: Same

File: src/main/java/com/example/carsharingservice/model/User.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.Table;
+import lombok.Data;
+
+@Data
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
```
Comment (AleksandrLts) [inline src/main/java/com/example/carsharingservice/model/User.java]: add unique property

File: src/main/java/com/example/carsharingservice/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final UserMapper mapper;
+
+    @PutMapping("/{id}/role")
+    public UserResponseDto updateRole(@PathVariable Long id, @RequestParam String role) {
+        User user = userService.get(id);
+        user.setRole(Role.valueOf(role));
```
Comment (AleksandrLts) [inline src/main/java/com/example/carsharingservice/controller/UserController.java]: you need to check if role already exist in Role class

**Review by AleksandrLts — CHANGES_REQUESTED** [review]
Gj, fix a little

**Review by AleksandrLts — APPROVED** [review]

**Review by marianapolishchuk2169 — APPROVED** [review]

**Pull request merged**